### PR TITLE
Code Insights: Assemble compute creation UI

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes part of https://github.com/sourcegraph/sourcegraph/issues/37965
Closes https://github.com/sourcegraph/sourcegraph/issues/37961

## Test plan
- Map picker should give the ability to pick one of the result map options
- If one or more series don't have type: diff or type: commit filters we should disable author and data options
- If we disable options that have been picked already we should reset the picker to the default value which is the repository group option. 

## App preview:

- [Web](https://sg-web-vk-assemble-compute-creation-ui.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-mvmnfmycyj.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/insights/pages/insights/creation/search-insight/utils/use-initial-values.ts [hunk 1]
```diff
@@ -25,8 +25,9 @@ export function useSearchInsightInitialValues(): UseInitialValuesResult {
 
     const urlParsedInsightValues = useMemo(() => decodeSearchInsightUrl(search), [search])
 
-    // Creation UI saves all form values in local storage to be able restore these
+    // Creation UI saves all form values in local storage to be able to restore these
     // values if page was fully refreshed or user came back from other page.
+    // eslint-disable-next-line no-restricted-syntax
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/pages/insights/creation/search-insight/utils/use-initial-values.ts:30]: Same as before. Please leave a comment as to why we are disabling this and not fixing it.

**Review by unclejustin — APPROVED** [review]
Just a couple of comments about disabling some eslint rules.

**Comment by vovakulikov** [comment]
FYI @AlicjaSuska I intentionally didn't add you to this PR since compute powered insight creation UI is still work in progress. I suggest let's review the whole page layout in final PR about the original issue about compute creation UI [here](https://github.com/sourcegraph/sourcegraph/issues/37965)

**Comment by vovakulikov** [comment]
I'm about to merge this PR. @AlicjaSuska I will assign you the PR about this page in order to conduct a UI review but still thanks for reviewing all ongoing PRs and watching that all of them are properly UI reviewed

**Pull request merged**